### ISSUE TITLE
BottomSheetLayout requires a child

### ISSUE THREAD

**Issue by schwiz** [Flipboard/bottomsheet#30]
If BottomSheetLayout doesn't have any children it will throw an exception when trying to show a sheet.
For now the work around is to throw in an empty view like below. 

    <com.flipboard.bottomsheet.BottomSheetLayout
        android:id="@+id/bottomsheet"
        android:fitsSystemWindows="true"
        android:layout_width="match_parent"
        android:layout_height="match_parent">
        <View
            android:layout_width="wrap_content"
            android:layout_height="wrap_content"/>
        </com.flipboard.bottomsheet.BottomSheetLayout>

**Comment by hzsweers** [comment]
This is intentional. What would be the use case for not showing a child view?

**Comment by schwiz** [comment]
My view is super complicated and basically BottomSheetLayout can't be the top level view. The only job of BottomSheetLayout is to show the sheet, there is no reason why my other none related content should be inside of it. In most cases that wouldn't hurt but it just doesn't work correctly in this particular case.

Here is a use case:

    <Frame>
         <FitsWindowsFrame>
             
         </FitsWindowFrame>
         <FullScreenVideoFrame/>
         <BottomSheetLayout/>
    </Frame>

Here what I want to do is allow the BottomSheetLayout to cover the fullscreen video if they hit its share button, but the bottom sheet can also be display from the content view when the video isn't full screen. Does this use case make sense?

**Comment by st028** [comment]
I'm not sure if I understand the difference between the fullscreen/non-fullscreen cases. Also, what isn't working as expected when you make the bottomsheet the root?

**Comment by schwiz** [comment]
I want BottomSheetLayout to fit the system windows but I don't want the top level view to fit the system windows.  I just had a thought though, perhaps the child sheetview can fit system windows and BottomSheetLayout doesn't need to.

**Comment by st028** [comment]
I think I understand now. You used BottomSheetLayout as the root for your view containing the video initially and in cases where the video was fullscreen, it worked as expected. But when the video was taking up only part of the screen, the bottomsheet would only show on the video. Is this correct?

So, in our usecases at Flipboard, we make BottomSheetLayout the root view of the activity content view that we want a bottomsheet in, allowing the bottomsheet to expand to fit the system window. We then define the rest of the layout as usual.

**Comment by st028** [comment]
Bottomsheet follows a similar pattern as the google's navigation drawer (https://developer.android.com/intl/ru/training/implementing-navigation/nav-drawer.html), in the respect that it must be the root of the content view in order to work as expected.

**Comment by schwiz** [comment]
My video logic actually detaches the SurfaceView of the video from its parent and re-attaches it to the full screen container FrameLayout when the device is rotated. This is so the video can be hosted inside a view that wouldn't fill up the screen when the device is rotated (ie it is inside the fit windows container, with a complex layout).

So my issue is/was if the share button was hit while in fullscreen mode the video would cover the sheet layout, because the sheetlayout was inside the fit system window layout. But if I pull the BottomSheetLayout to contain everything then part of the sheet is covered up by the system ui.

I worked around this for now by just manually setting the padding of sheetviews at runtime to match the system ui padding. 

I guess this would be a good argument for switching over to the CoordinatorLayout.Behavior rather than following the nav drawer pattern as discussed here. https://github.com/Flipboard/bottomsheet/issues/9  

Closing this ticket because as you say the layout is working as intended.
